client/express-checkout/index.js

```diff
@@ -337,10 +402,14 @@ jQuery( ( $ ) => {
 			} else if ( wcpayExpressCheckoutParams.is_product_page ) {
 				wcpayECE.startExpressCheckoutElement( {
 					mode: 'payment',
-					total: wcpayExpressCheckoutParams.product.total.amount,
-					currency: 'usd',
+					total: getExpressCheckoutData( 'product' )?.total.amount,
+					currency: getExpressCheckoutData( 'product' )?.currency,
 					requestShipping:
-						wcpayExpressCheckoutParams.product.needs_shipping,
+						getExpressCheckoutData( 'product' )?.needs_shipping ??
+						false,
+					requestPhone:
+						getExpressCheckoutData( 'checkout' )
+							?.needs_payer_phone ?? false,
 					displayItems:
 						wcpayExpressCheckoutParams.product.displayItems,
```

Comment:
Should we also update this one?
```javascript
getExpressCheckoutData( 'product' )?.displayItems,
```

Reply:
I'd rather not introduce a bunch of unrelated changes to the PR, so let's keep pre-existing cases as they are and use `getExpressCheckoutData` for new code.

I'm not super tied to that opinion though, happy to make changes where we deem it appropriate.